trace/trace.go

```diff
@@ -469,44 +313,16 @@ func (s *Span) lazyPrintfInternal(attributes []Attribute, format string, a ...in
 	s.mu.Unlock()
 }
 
-// LazyPrintfWithAttributes adds an annotation with attributes to the current span.
-//
-// The format string is evaluated with its arguments only when the annotation text is needed.
-func LazyPrintfWithAttributes(ctx context.Context, attributes []Attribute, format string, a ...interface{}) {
-	s, ok := ctx.Value(contextKey{}).(*Span)
-	if !ok {
-		return
-	}
-	s.LazyPrintfWithAttributes(attributes, format, a...)
-}
-
-// LazyPrintfWithAttributes adds an annotation with attributes.
+// Logf adds an annotation with attributes.
 //
 // The format string is evaluated with its arguments only when the annotation text is needed.
-func (s *Span) LazyPrintfWithAttributes(attributes []Attribute, format string, a ...interface{}) {
+func (s *Span) Logf(attributes []Attribute, format string, a ...interface{}) {
```

Comment:
I think you made the right choice in keeping the lazy one, but you should take care to mention how that can cause problems. I think it could be as simple as ".... only when the annotation text is needed, so any arguments should not be modified subsequently" or something like that.

Reply:
`lazyPrintfInternal` formats the string so there isn't an opportunity for arguments to be modified subsequently in the current implementation. I assume you mean we should leave open the door to allow future implementation changes that would retain the arguments longer? If that's the case, I don't know if we should that - it seems like asking too much of the developer to think that much about a logging function.

Reply:
(Unless of course the contemplated implementation change is well-understood, imminent and unavoidable)

Reply:
I thought that format call was just a placeholder, and the intent was always to do defer the formatting for longer. Currently there isn't a real difference between the lazy and non-lazy cases, I believe.